src/flow/maximum_flow.jl

```diff
@@ -60,20 +60,7 @@ Requires arguments:
 - capacity_matrix::AbstractArray{T,2}     # input capacity matrix
 """
 
-function residual(
-    flow_graph::DiGraph                     # the input graph
-    )
-
-    n = nv(flow_graph)
-    residual_graph = copy(flow_graph)       # make a copy of the input graph
-    for (u,v) in edges(flow_graph)
-        if !has_edge(flow_graph, v, u)      # create reverse edge
-            add_edge!(residual_graph, v, u)
-        end
-    end
-
-    return residual_graph
-end
+residual(flow_graph::DiGraph) = DiGraph(Graph(flow_graph))
```

Comment:
It is, in fact, related to the PR. If you look at the original code, it called `add_edge!` after a `has_edge` test - this needed to be changed in any case. As it turns out, rewriting residual in this way also made it more efficient and performant.